src/components/Dialog.tsx

```diff
@@ -94,6 +97,10 @@ export default class Dialog extends Component<DialogProps, DialogState> {
     return enabledSources;
   };
 
+  handleTotalImageCount = (totalImageCount: number) => {
+    this.setState({ totalImageCount });
+  };
+
```

Comment:
Maybe we can track the page count total in state as well?
So that his fn looks more like:

```js
const current = this.state.page.current;
this.setState( {
  totalImageCount,
  page: {
    current,
    total: Math.ceil(totalImageCount / 18} 
  })
```

Probably best to tackle this in a subsequent PR however.